docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,225 @@
+
+## Adoption Process Diagram
+
+![sequence-diagram](./images/adoption-sequence-diagram.png)
+# Potential Alternatives
+
+## Resource Adoption Annotation
+
+The existing custom resource definitions (every service resource) would accept an additional annotation that the manager interprets as an indication that the resource specification should be ignored and instead that it should be formed through adoption. For any resource marked with this annotation, upon creation, the manager would attempt to describe the AWS resource associated with the annotated value. If none such resource exists, the manager would update the state of the custom resource into a `AdoptionFailed` terminal state. If the resource does exist, the manager would continue the follow the same adoption process as defined in the [Proposed Implementation](#proposed-implementation) section.
+
+### Example Specification
+
+**An example adopted S3 resource**
+
+```
+apiVersion: s3.services.k8s.aws/v1alpha1
+kind: Bucket
+metadata:
+  annotations:
+    services.k8s.aws/arn: arn:aws:s3:::my-existing-bucket
+  ...
+spec:
+   ...
+```
+
+### Notable Pitfalls
+
+In the event that the user is attempting to restore resources created in a previous installation of ACK (in the event of an etcd corruption, for example), each manifest will need to be updated with the adoption annotation. Therefore the restoration process would require modifying desired state files (k8s manifests) with actual state fields (the created ARN/ID fields). At the least this introduces a temporary change in k8s manifests - add ARNs to manifests,  apply and adopt, delete ARNs from manifests. More so, it could introduce private field information (generated public API IDs, RDS instance ARNs, etc.) into the manifests repository.
+
+Another pitfall is that, given this solution aims to re-use the existing custom resource definitions, all adopted resources would need to provide a specification body that contains all required fields. Every service contains a distinct set of required values, and as such the user would need to modify every adopted resource type with a distinct set of valid dummy values. Assuming a user provides dummy values for these fields, they would subsequently be overwritten by the manager logic when updating the resource.
+
+# In Scope
+
+* Declaration of the new `AdoptedResource` CRD
+* Updates to generator for service-specific configuration
+* Code paths for service managers to receive informer updates
+* Unit and E2E tests with a specific ACK controller (e.g. S3 buckets)
+* Verbose logging for adoption lifecycle updates
+
+# Out of Scope
+
+* Integration of adopted resource types for every ACK controller
+* E2E tests for every ACK controller
+* Tooling for generating `AdoptedResource` CRs
```

Comment:
To clarify, in order to adopt a resource, will a user need to add  "an additional annotation" (line 134) on the existing ACK manifest and to create the corresponding `AdoptedResource` CRs?

Reply:
The additional annotation is an alternative we explored but found was not suitable. Instead, the user only needs to create an `AdoptedResource` CR.